src/mongocxx/include/mongocxx/v1/indexes.hpp

```diff
@@ -0,0 +1,64 @@
+// Copyright 2009-present MongoDB, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <mongocxx/v1/indexes-fwd.hpp>
+
+//
+
+#include <mongocxx/v1/detail/prelude.hpp>
+
+namespace mongocxx {
+namespace v1 {
+
+///
+/// Support for MongoDB indexes.
+///
+/// @note Not to be confused with Atlas Search indexes or Atlas Vector Search indexes.
+///
+/// @see
+/// - [Indexes (MongoDB Manual)](https://www.mongodb.com/docs/manual/indexes/)
+///
+/// @attention This feature is experimental! It is not ready for use!
+///
+class indexes {
+   public:
+    ///
+    /// A description of a MongoDB index.
+    ///
+    /// @see
+    /// - [Indexes (MongoDB Manual)](https://www.mongodb.com/docs/manual/indexes/)
+    ///
+    /// @attention This feature is experimental! It is not ready for use!
+    ///
+    class model {};
+
+    ///
+    /// Options for @ref mongocxx::v1::indexes.
+    ///
+    /// @attention This feature is experimental! It is not ready for use!
+    ///
+    class options {};
```

Comment:
`v_noabi::options::index` seems to be nearly-completely obsoleted by `v_noabi::options::index_view`. It's only current usage is in:

- Isolated test and example code: easily replaceable.
- `v_noabi::gridfs::bucket::create_indexes_if_nonexistent`: private API, easily replaceable.
- `v_noabi::options::index::operator bsoncxx::v_noabi::document::view_or_value`: to support the deprecated `*_storage_options` API (not included in `v1`).

Therefore, `v1::indexes::options` will merge and fulfill the role of both the old `index` and `index_view` options classes.